lib/linguist/heuristics.yml

```diff
@@ -709,6 +709,10 @@ disambiguations:
   - language: Markdown
     # Markdown syntax for scdoc
     pattern: '^#+\s+(NAME|SYNOPSIS|DESCRIPTION)'
+- extensions: ['.script']
+  rules:
+  - language: URScript
+    pattern: '^\s*def\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*\(([^)]*)\)\s*:(((?:(?!^\send($|\s))[\s\S])*)\send($|\s))'
```

Comment:
Thanks for the improvement, however this now has a problem in that it's no longer run linearly which means it'll likely have performance issues:

```ruby
irb(main):002> Regexp.linear_time?(/^\s*def\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*\(([^)]*)\)\s*:(((?:(?!^\send($|\s))[\s\S])*)\send($|\s))/)
=> false
```

We're trying to improve the regexes in this file to improve performance and prevent timing out (we catch these ourselves but really shouldn't) so we're aiming for all regexes to have linear execution whilst also being immune from ReDos. [^1]

/cc @Alhadis

[^1]: I'm slowly doing this in a separate branch but ultimately this will be enforced so we may as well get it right now rather than down the road when it's enforced.

Reply:
It makes a lot of sense. Thank you for insight!